### PR TITLE
xEQUAL ut

### PR DESCRIPTION
**Pull request by shargon**
For same types, i want to add more tests between different types (in the future)

### CODE REVIEW

File: src/neo-vm/Unsafe.cs [hunk 1]
```diff
@@ -45,8 +45,8 @@ public static void MemoryCopy(byte[] src, int srcOffset, byte[] dst, int dstOffs
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static bool NotZero(byte[] x)
         {
-            if (x is null)
-                throw new ArgumentNullException(nameof(x));
+            if (x is null) throw new ArgumentNullException(nameof(x));
+
```
Comment (erikzhang) [inline src/neo-vm/Unsafe.cs]: Unnecessary change.

Comment (shargon) [reply src/neo-vm/Unsafe.cs]: otherwise i need to do a specific test for hit this line

File: src/neo-vm/Unsafe.cs [hunk 2]
```diff
@@ -1,6 +1,7 @@
 ﻿using System;
 using System.Runtime.CompilerServices;
 
+[assembly: InternalsVisibleTo("neo-vm.Tests")]
```
Comment (erikzhang) [inline src/neo-vm/Unsafe.cs]: We usually add this line to `Properties/AssemblyInfo.cs`.

Comment (shargon) [reply src/neo-vm/Unsafe.cs]: There is AssemblyInfo in dotnet core?

Comment (erikzhang) [reply src/neo-vm/Unsafe.cs]: https://github.com/neo-project/neo/blob/master/neo/Properties/AssemblyInfo.cs

File: src/neo-vm/Unsafe.cs [hunk 3]
```diff
@@ -1,70 +1,70 @@
...
-                {
-                    if (*xbp != 0) return true;
-                    xbp++;
-                }
-            }
-            return false;
-        }
-    }
-}
+﻿using System;
```
Comment (shargon) [inline src/neo-vm/Unsafe.cs]: I don't know understand why github mark all this file as changed, when is the same

Comment (diskooooo) [reply src/neo-vm/Unsafe.cs]: > I don't know understand why github mark all this file as changed, when is the same

Most probably character encoding: it could be due a change from DOS to UNIX line breaks, for example.

Comment (shargon) [reply src/neo-vm/Unsafe.cs]: But i copy the same file, and git doesn't recognize any change.
